containers/unit_tests/TestVector.hpp

```diff
@@ -52,6 +52,125 @@ namespace Test {
 
 namespace Impl {
 
+template <typename Scalar, class Device>
+struct test_vector_insert {
+  typedef Scalar scalar_type;
+  typedef Device execution_space;
+
+  template <typename Vector>
+  void run_test(Vector& a) {
+    int n = a.size();
+
+    auto it = a.begin();
+    it += 15;
+    ASSERT_EQ(*it, scalar_type(1));
+
+    auto it_return = a.insert(it, scalar_type(3));
+    ASSERT_EQ(a.size(), n + 1);
+    ASSERT_EQ(std::distance(it_return, a.begin() + 15), 0);
+
+    it = a.begin();
+    it += 17;
+// Looks like some std::vector implementations do not have the restriction
+// right on the overload taking three iterators, and thus the following call
+// will hit that overload and then fail to compile.
+#if defined(KOKKOS_COMPILER_INTEL) && (1700 > KOKKOS_COMPILER_INTEL)
+// And at least GCC 4.8.4 doesn't implement vector insert correct for C++11
+// Return type is void ...
+#if (__GNUC__ < 5)
+    a.insert(it, typename Vector::size_type(n + 5), scalar_type(5));
```

Comment:
Okay, here's where you're doubling the size. I missed that.

Reply:
@crtrott Please comment to make it obvious that this will exceed the extra storage space.